test/fail_compilation/test17284.d

```diff
@@ -0,0 +1,20 @@
+/* REQUIRED_ARGS:
+TEST_OUTPUT:
+---
+fail_compilation/test17284.d(16): Error: field `U.c` cannot access pointers in `@safe` code that overlap other fields
+pure nothrow @safe void(U t)
```

Comment:
How about actually verifying that attribute inference actually considers this function `@system`?
Also In the bug report me and H. S. Teoh posted several more test cases, which doesn't seem to be addressed here.

Reply:
> How about actually verifying that attribute inference considers this function @system?

The message does that.

> Also In the bug report me and H. S. Teoh posted several more test cases, which doesn't seem to be addressed here.

This is the root cause. If you find more cases after this one is pulled, please point them out.